### ISSUE TITLE
SmartGit doesn't show it's main window

### ISSUE THREAD

**Issue by aksdb** [hyprwm/Hyprland#1516]
SmartGit (a git client written in Java using SWT) doesn't show a window when run in Hyprland. If it shows a dialog (after a crash for example), the dialog shows up, but the main window doesn't. The process keeps running and it's logs indicates, that it thinks it has a window. `hyprctl clients` doesn't show a window either, though.

Running it on *sway* works, so it's probably not a generic aversion to wayland or tiling wms.

I tried a directly installed version and the flatpak version - both on ArchLinux.

Additional Info: after many many many restarts it showed the main window once, but then failed to show one of its modal child windows, so I was stuck and couldn't continue (since I couldn't close the non-showing modal window).

**Comment by sirris-bat** [comment]
Seeing the same behaviour in the Warframe launcher. After a semi-recent change the client is invisible and is not represented in `hyprctl clients`. Is certainly running however, as it is playing the ambient music.

**Comment by vaxerski** [comment]
> After a semi-recent change

if this change is a hyprland change, please do a `git bisect`

**Comment by Wibjarm** [comment]
Just tested this with the Warframe launcher, 668842124067f9b6513edf1c28478bec12d709a0 is the last working commit, fcf1bafb098c8f756550195269dcd95202124277 is the first broken commit.

**Comment by sirris-bat** [comment]
> Just tested this with the Warframe launcher, [6688421](https://github.com/hyprwm/Hyprland/commit/668842124067f9b6513edf1c28478bec12d709a0) is the last working commit, [fcf1baf](https://github.com/hyprwm/Hyprland/commit/fcf1bafb098c8f756550195269dcd95202124277) is the first broken commit.

Thanks, was struggling to get it to build!

**Comment by aksdb** [comment]
> Just tested this with the Warframe launcher, [6688421](https://github.com/hyprwm/Hyprland/commit/668842124067f9b6513edf1c28478bec12d709a0) is the last working commit, [fcf1baf](https://github.com/hyprwm/Hyprland/commit/fcf1bafb098c8f756550195269dcd95202124277) is the first broken commit.

In that case it might be a different problem than mine, because I also encountered it in [v0.21.0beta](https://github.com/hyprwm/Hyprland/tree/v0.21.0beta).

**Comment by aksdb** [comment]
Apparently it can be reproduced with Eclipse IDE as well, when launching it using `GDK_BACKEND=x11 ./eclipse`. The workspace selection dialog and the splash screen show up, but the main window doesn't.

**Comment by vaxerski** [comment]
can you now check with 89e7d812c80038083acdca85a5ba08ad97b2d0ea whether it appears in hyprctl clients?

**Comment by aksdb** [comment]
@vaxerski The windows show up in the list now. For example like this:

```
Window 704fb270 -> :
	mapped: 0
	hidden: 0
	at: 0,0
	size: 0,0
	workspace: -1 ()
	floating: 0
	monitor: -1
	class: unmanaged X11
	title: SmartGit 22.1.4 
	pid: 5511
	xwayland: 1
	pinned: 0
	fullscreen: 0
	fullscreenmode: 0
	fakefullscreen: 0
	grouped: 0
	swallowing: 0

Window 704fdfc0 -> :
	mapped: 0
	hidden: 0
	at: 0,0
	size: 0,0
	workspace: -1 ()
	floating: 0
	monitor: -1
	class: unmanaged X11
	title: Welcome to SmartGit 
	pid: 5511
	xwayland: 1
	pinned: 0
	fullscreen: 0
	fullscreenmode: 0
	fakefullscreen: 0
	grouped: 0
	swallowing: 0
```

"unmanaged X11" sounds suspicious.

**Comment by vaxerski** [comment]
mhm. I have an idea.

**Comment by vaxerski** [comment]
fixed with 7e523e4d5e526bdf5ab86a8f4fe37efeef356707 :)

**Comment by aksdb** [comment]
@vaxerski Can confirm. The windows show up fine now. Thank you very much :smiley:
